modules/python/python-persist.c

```diff
@@ -22,7 +22,41 @@
 
 #include "python-persist.h"
 #include "python-helpers.h"
+#include "syslog-ng.h"
 #include "driver.h"
+#include "mainloop.h"
+#include "compat/compat-python.h"
+
+#include <structmember.h>
+
+#define SUBKEY_DELIMITER "##"
+
+typedef enum
+{
+  ENTRY_TYPE_STRING,
+  ENTRY_TYPE_LONG,
+  ENTRY_TYPE_MAX
+} EntryType;
+
+typedef struct
+{
+  guint8 type;
+  gchar data[];
+} Entry;
+
+static PyObject *
+entry_to_pyobject(guint8 type, gchar *value)
+{
+  switch (type)
+    {
+    case ENTRY_TYPE_STRING:
+      return _py_string_from_string(value, -1);
```

Comment:
We might need to support binary data here (bytes). 

At least I am not sure systemd for instance doesn't give us binary bookmarks.

Reply:
You are right. I added a new patch to the top. From now, we can store string, number and bytes.